Sources/ApolloCodegenLib/ApolloCodegenConfiguration.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+
+/// A configuration object that defines behavior for code generation.
+public struct ApolloCodegenConfiguration {
+  /// Specify the input files required for code generation.
+  public struct FileInput {
+    /// Path to the GraphQL schema file. Can be in JSON or SDL format.
+    public let schema: URL
+    /// Glob of files to search for GraphQL operations. This should be used to find queries and any client schema extensions. Defaults to `./**/*.graphql`, which will search for `.graphql` files throughout all subfolders of the folder where the script is run.
+    public let glob: String
+    /// [optional] Path to an operation id JSON map file. If specified, also stores the operation IDs (hashes) as properties on operation types. Defaults to `nil`.
+    public let operationIDs: URL?
+  }
+
+  /// Configure the folder structure of the generated files.
+  public struct FileOutput {
+    /// The location structure for the schema types files.
+    public let schemaTypes: SchemaTypesFileOutput
+    /// The location structure for the operation object files.
+    public let operations: OperationsFileOutput
+  }
+
+  /// Defines the location structure for the schema types files.
+  public struct SchemaTypesFileOutput {
+    /// Compatible dependency manager automation.
+    public enum DependencyManagerAutomation {
+      /// No automation. The generated schema types files must be manually added to the main application target.
+      case includedInTarget
+      /// Generate a podspec file.
+      case cocoaPods
+      /// Generate a cartfile.
+      case carthage
+      /// Generate a package.swift file.
+      case swiftPackageManager
+    }
+
+    /// The namespace to use for generated operation objects. When combined with `.includedInTarget` this should be the main target name. When used with any other dependency manager this should be the name of the new shared module that will be created.
```

Comment:
> When combined with `.includedInTarget` this should be the main target name.

That's not right.

When combined with `.includedInTarget` this will be the name of the namespace `enum` that we put the schema types in. It should definitely never be the name of your application target.